packages/xrpl/snippets/src/bridgeTransfer.ts

```diff
@@ -0,0 +1,166 @@
+    try {
+      initialBalance = await issuingClient.getXrpBalance(wallet2.classicAddress)
+      console.log(
+        `Wallet ${wallet2.classicAddress} has been funded with a balance of ${initialBalance} XRP`,
+      )
+      break
+    } catch (_error) {
+      ledgersWaited += 1
+      // eslint-disable-next-line max-depth -- needed here
+      if (ledgersWaited === MAX_LEDGERS_WAITED) {
+        throw Error('Destination account creation via the bridge failed.')
+      }
+    }
+  }
+
+  console.log(
+    `Transferring funds from ${wallet1.classicAddress} on the locking chain to ` +
+      `${wallet2.classicAddress} on the issuing_chain...`,
+  )
+
+  // Fetch the claim ID for the transfer
+  console.log('Step 1: Fetching the claim ID for the transfer...')
+  const claimIdTx: XChainCreateClaimID = {
+    TransactionType: 'XChainCreateClaimID',
+    Account: wallet2.classicAddress,
+    XChainBridge: bridge,
+    SignatureReward: bridgeData.SignatureReward,
+    OtherChainSource: wallet1.classicAddress,
+  }
+  const claimIdResult = await issuingClient.submitAndWait(claimIdTx, {
+    wallet: wallet2,
+  })
+
+  // extract new claim ID from metadata
+  const nodes = (claimIdResult.result.meta as TransactionMetadata).AffectedNodes
+  const createdNodes = nodes.filter((node) =>
+    Object.keys(node).includes('CreatedNode'),
+  )
+  const claimIdsLedgerEntries = (createdNodes as CreatedNode[]).filter(
+    (node) => node.CreatedNode.LedgerEntryType === 'XChainOwnedClaimID',
+  )
+  if (claimIdsLedgerEntries.length !== 1) {
+    // This error should never hit and is just a sanity check
+    throw Error('Wallet has more than one claim ID')
+  }
+  const xchainClaimId = claimIdsLedgerEntries[0].CreatedNode.NewFields
+    .XChainClaimID as string
+
+  console.log(`Claim ID for the transfer: ${xchainClaimId}`)
+
+  // XChainCommit
+  console.log('Step 2: Locking the funds on the locking chain...')
+
+  const commitTx: XChainCommit = {
+    TransactionType: 'XChainCommit',
+    Account: wallet1.classicAddress,
+    Amount: xrpToDrops(1),
+    XChainBridge: bridge,
+    XChainClaimID: xchainClaimId,
+    OtherChainDestination: wallet2.classicAddress,
+  }
+  const commitResult = await lockingClient.submitAndWait(commitTx, {
+    wallet: wallet1,
+  })
+  console.log(commitResult)
+
+  // wait for the attestations to go through
+  ledgersWaited = 0
+  while (ledgersWaited < MAX_LEDGERS_WAITED) {
+    await sleep(LEDGER_CLOSE_TIME)
+    const currentBalance = await issuingClient.getXrpBalance(
+      wallet2.classicAddress,
+    )
+    console.log(initialBalance, currentBalance)
+    if (parseFloat(currentBalance) > parseFloat(initialBalance)) {
+      console.log('Transfer is complete')
+      console.log(
+        `New balance of ${wallet2.classicAddress} is ${currentBalance} XRP`,
+      )
+      break
+    }
+
+    ledgersWaited += 1
+    if (ledgersWaited === MAX_LEDGERS_WAITED) {
+      throw Error('Bridge transfer failed.')
+    }
+  }
```

Comment:
This logic shows up twice, and I think will be fairly common for cross-chain transactions. Can we turn this into a helper method like `client.xChainSubmitAndWait(...)` or something like that? (Then `MAX_LEDGERS_WAITED` becomes an optional parameter)

Reply:
It seems a bit complex to put that logic into the library itself, which is why I just wrote a snippet for it. Devs can copy-paste it into their own code.